src/Common/ItechArt.Repositories/UnitOfWork.cs

```diff
@@ -0,0 +1,70 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using ItechArt.Repositories.Abstractions;
+using Microsoft.EntityFrameworkCore;
+
+namespace ItechArt.Repositories;
+
+public class UnitOfWork<TContext> : IUnitOfWork
+    where TContext : DbContext
+{
+    private IDictionary<Type, object> _repositories;
+    private bool _disposed;
+
+
+    public TContext DbContext { get; }
+
+
+    public UnitOfWork(TContext dbContext)
+    {
+        DbContext = dbContext
+            ?? throw new ArgumentNullException(nameof(dbContext));
+    }
+
+
+    public IRepository<TEntity> GetRepository<TEntity>()
+        where TEntity : class, IEntityId
+    {
+        if (_repositories == null)
+        {
+            _repositories = new Dictionary<Type, object>();
+        }
+
+        var type = typeof(TEntity);
+
+        if (!_repositories.ContainsKey(type))
+        {
+            _repositories[type] = new Repository<TEntity>(DbContext);
```

Comment:
Add support for custom repository types:
1. UoW should support configuring entity type/repository type mapping
2. UoW should be able to create custom repository types for entities configured in p.1 above